src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -230,11 +230,38 @@ def read_file_content(file_path, allow_binary=False):
     raise CLIError('Failed to decode file {} - unknown decoding'.format(file_path))
 
 
+def preprocess_json(original):
+    lines = original.splitlines()
+    for line_index, line in enumerate(lines):
+        if not ("#" in line or "//" in line):
```

Comment:
Are you sure # works as a comment?  PowerShell doesn't allow them, although I'm not sure if the backend would allow them if they got past PowerShell.

Reply:
Well, I just followed the same idea as in **commentjson**.
So currently I allow both "//" and "#"
If PS doesnt' allow "#" then probably it just won't appear in any of the templates, so it's not a problem.

Reply:
Lidong says the backend doesn't support # as a comment

Reply:
It does seem odd to support "#" comments, as this will be flagged as an error in vs code and will result in an error when it is sent to the ARM backend.